scene/resources/skeleton_modification_3d.h

```diff
@@ -0,0 +1,253 @@
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef SKELETONMODIFICATION3D_H
+#define SKELETONMODIFICATION3D_H
+
+#include "scene/3d/skeleton_3d.h"
+
+///////////////////////////////////////
+
+class Skeleton3D;
+class SkeletonModification3D;
+
+class SkeletonModificationStack3D : public Resource {
+	GDCLASS(SkeletonModificationStack3D, Resource);
+	friend class Skeleton3D;
+	friend class SkeletonModification3D;
+
+protected:
+	static void _bind_methods();
+	void _get_property_list(List<PropertyInfo> *p_list) const;
+	bool _set(const StringName &p_path, const Variant &p_value);
+	bool _get(const StringName &p_path, Variant &r_ret) const;
+
+public:
+	Skeleton3D *skeleton;
+	bool is_setup;
+	bool enabled;
+	float strength;
+
+	Vector<Ref<SkeletonModification3D>> modifications;
+	int modifications_count;
+
+	void setup();
+	void execute();
+
+	void enable_all_modifications(bool p_enable);
+	Ref<SkeletonModification3D> get_modification(int p_mod_idx) const;
+	void add_modification(Ref<SkeletonModification3D> p_mod);
+	void delete_modification(int p_mod_idx);
+	void set_modification(int p_mod_idx, Ref<SkeletonModification3D> p_mod);
+
+	void set_modification_count(int p_count);
+	int get_modification_count() const;
+
+	void set_skeleton(Skeleton3D *p_skeleton);
+	Skeleton3D *get_skeleton() const;
+
+	bool get_is_setup() const;
+
+	void set_enabled(bool p_enabled);
+	bool get_enabled() const;
+
+	void set_strength(float p_strength);
+	float get_strength() const;
+
+	SkeletonModificationStack3D();
+};
+
+///////////////////////////////////////
+
+class SkeletonModification3D : public Resource {
+	GDCLASS(SkeletonModification3D, Resource);
+	friend class Skeleton3D;
+
+protected:
+	static void _bind_methods();
+
+	SkeletonModificationStack3D *stack;
+
+	bool enabled;
+	bool is_setup;
+
+public:
+	virtual void execute();
+	virtual void setup_modification(SkeletonModificationStack3D *p_stack);
+
+	void set_enabled(bool p_enabled);
+	bool get_enabled();
+
+	SkeletonModification3D();
+};
+
+///////////////////////////////////////
+
+class SkeletonModification3D_LookAt : public SkeletonModification3D {
+	GDCLASS(SkeletonModification3D_LookAt, SkeletonModification3D);
+
+private:
+	String bone_name;
```

Comment:
Useful to be StringName for performance reasons of a pointer compare vs string compare?

Reply:
`bone_name` is just used to get the bone index using the `find_bone` function in Skeleton3D, and in a couple conditions. The function returns a String, so I figured I should store it as such. Would storing it as a StringName be more performant?
I tested with switching it to a StringName and it compiled okay, but it wasn't properly getting the bone index from the `find_bone` function. To be fair though, I only changed the type to a StringName and fixed a couple errors, so it is possible that something needs adjusting for it to work as a StringName and I missed that.

Reply:
Do you need to compare the whole thing or character by character?

For comparison of whole strings, a StringName is faster.

Reply:
Looking at the code, it looks like the only comparison is for checking if it is an empty string (`if bone_name == ""` or equivalent), and this is only in one of the two modifications (LookAt).

Thinking about it, I can probably remove the condition altogether from the modifications, making it just for storage and use in the `find_bone` function.